dask/array/core.py

```diff
@@ -3508,6 +3508,14 @@ def concatenate3(arrays):
 
     advanced = max(core.flatten(arrays, container=(list, tuple)),
                    key=lambda x: getattr(x, '__array_priority__', 0))
+
+    if not isinstance(advanced, np.ndarray) and hasattr(advanced, '__array_function__'):
```

Comment:
What if we have `__array_function__` but the NumPy version isn't advanced enough?

Reply:
I think some sort of way exists to detect if the protocol exists and is enabled. This is a flaw I spotted, but there's also the issue of the environment variable along with the version.

cc @shoyer is there an easy way to check for an (enabled) `__array_function__` protocol?

Reply:
We can probably add a check at import time.

Reply:
You could look at the NumPy version and environment variables. With 1.17 it will always be enabled. With 1.16 you need to opt in the with env variable.

Reply:
NEP-18 is still provisional, even in 1.17 right?

Reply:
Yes, I suppose so.

Reply:
Then that route seems kinda flaky... @mrocklin If you want I could write a simple test to see if `__array_function__` exists on import and expose it. Maybe such a thing is useful in NumPy itself?

Reply:
Yes, I think that having a small utility function `is_nep18_active` or something similar would be useful here.  If we find it valuable in Dask then maybe that's good evidence to push upstream to NumPy.

Reply:
Opened a new PR (#4675) to that effect.